docs/Tools/Barstools.rst

```diff
@@ -3,3 +3,92 @@ Barstools
+The ``fallbacksynflops`` value causes MacroCompiler to compile all possible memories to technology macros but when unable to do so to use flip flops to implement the remaining memories.
+The final and default value, ``compileavailble``, instructs MacroCompiler to compile all memories to the technology macros and do nothing if it is unable to map them.
+
+Most of the rest of the options are used to control where different inputs and outputs are expected and produced.
+The option ``--macro-conf`` is the file that contains the set of input ``SeqMem`` configuartions to map in the ``.conf`` format described above.
+The option ``--macor-mdf`` also describes the input ``SeqMem`` s but is instead in the ``.mdf`` format.
+The option ``--library`` is an ``.mdf`` description of the available technology macros that can be mapped to.
+This file could be a list of fixed size memories often refered to as a cache of macros, or a description of what size memories could be made available through some technology specific process (usually an SRAM compiler), or a mix of both.
+The option ``--use-compiler`` instructs MacroCompiler that it is allowed to use any compilers listed in the ``--library`` specification.
+If this option is not set MacroCompiler will only map to macros directly listed in the ``--library`` specification.
+The ``--verilog`` option specifies where MacroCompiler will write the verilog containing the new technology mapped memories.
+The ``--firrtl`` option similarly specifies where MacroCompiler will write the FIRRTL that will be used to generate this verilog.
+This option is optional and no FIRRTL will be emitted if it is not specified.
+The ``--hammer-ir`` option specifies where MacroCompiler will write the details of which macros need to be generated from a technology compiler.
+This option is not needed if ``--use-compiler`` is not specified.
+This file can then be passed to HAMMER to have it run the technology compiler producing the associated macro collateral.
+The ``--cost-func`` option allows the user to specify a different cost function for the mapping task.
+Because the mapping of memories is a multi-dimensional space spanning performance, power, and area, the cost function setting of MacroCompiler allows the user to tune the mapping to their preference.
+The default option is a reasonable heurstic that attempts to minimize the number of technology macros instantiated per ``SeqMem`` without wasting too many memory bits.
+The ``--cost-param`` option allows the user to specify parameters to pass to the cost function if the cost function supports that.
+The ``--force-synflops [mem]`` options allows the user to override any heuristics in MacroCompiler and force it to map the given memory to flip-flops.
+Likewise, the ``--force-compile [mem]`` option allows the user to force MacroCompiler to map the given ``mem`` to a technology macro.
+
```

Comment:
Would it be useful to also list the short-form options (e.g. `-u` instead of `--use-compiler`)?

Reply:
I thought this would be unnecessary.